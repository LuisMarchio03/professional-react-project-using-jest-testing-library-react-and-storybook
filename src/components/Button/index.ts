import { styled } from '../../styles'

export const Button = styled('button', {
  fontSize: '$md',
  fontWeight: 600,
  textTransform: 'uppercase',
  padding: '12px 36px',
  cursor: 'pointer',

  variants: {
    variant: {
      primary: {
        backgroundColor: '$yellow500',
        border: '2px solid $yellow500',
        color: '$gray600',
      },
      secondary: {
        backgroundColor: 'transparent',
        border: '2px solid $yellow500',
        color: '$yellow500',
      },
    },
  },
})
