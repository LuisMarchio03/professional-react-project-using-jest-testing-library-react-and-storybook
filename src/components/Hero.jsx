export const Hero = ({ title, children }) => {
  return (
    <>
      <div>
        <div>
          <h1>{title}</h1>
          {children}
        </div>
      </div>
    </>
  )
}
