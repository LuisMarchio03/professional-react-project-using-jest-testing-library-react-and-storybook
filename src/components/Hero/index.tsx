import React from 'react'
import { Container } from './styles'

export interface HeroProps {
  image: string
  children: React.ReactNode
}

export const Hero = ({ image = '', children }: HeroProps) => {
  return (
    <>
      <Container
        css={{
          background: image,
        }}
        data-testid="hero"
      >
        <div>{children}</div>
      </Container>
    </>
  )
}
