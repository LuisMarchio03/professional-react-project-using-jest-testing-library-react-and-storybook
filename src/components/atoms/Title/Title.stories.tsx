import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'

import { Title } from '.'

import { globalStyles } from '../../../styles/global'
globalStyles()

export default {
  title: 'Components/Title',
  component: Title,
} as ComponentMeta<typeof Title>

const Template: ComponentStory<typeof Title> = (args) => <Title {...args} />

export const TitleH1 = Template.bind({})

TitleH1.args = {
  children: (
    <>
      Title <strong>h1</strong>
    </>
  ),
}

export const TitleH2 = Template.bind({})
TitleH2.args = {
  children: (
    <>
      Title <strong>h2</strong>
    </>
  ),
}

export const TitleH3 = Template.bind({})
TitleH3.args = {
  children: (
    <>
      Title <strong>h3</strong>
    </>
  ),
}

export const TitleH4 = Template.bind({})
TitleH4.args = {
  children: (
    <>
      Title <strong>h4</strong>
    </>
  ),
}

export const TitleH5 = Template.bind({})
TitleH5.args = {
  children: (
    <>
      Title <strong>h5</strong>
    </>
  ),
}

export const TitleH6 = Template.bind({})
TitleH6.args = {
  children: (
    <>
      Title <strong>h6</strong>
    </>
  ),
}
