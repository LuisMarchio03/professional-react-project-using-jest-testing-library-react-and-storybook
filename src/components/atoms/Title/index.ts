import { styled } from '../../../styles'

export const Title = styled('h1', {
  position: 'relative',
  // fontWeight: 700,
  letterSpacing: '0.05rem',
  marginBottom: '25px',
  paddingBottom: '25px',

  borderBottom: '1px solid rgba(255,255,255,0.2)',

  '&::after': {
    content: '""',
    position: 'absolute',
    left: '0',
    bottom: '-3px',
    backgroundColor: '$yellow500',
    width: '70px',
    height: '5px',
  },

  strong: {
    color: '$yellow500',
  },
})
