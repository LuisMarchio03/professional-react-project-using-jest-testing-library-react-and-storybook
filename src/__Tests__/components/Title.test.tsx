import React from 'react'
import { render } from '@testing-library/react'

import { Title } from '../../components/Title'

test('match snapshot', () => {
  const { asFragment } = render(<Title>Title</Title>)
  expect(asFragment()).toMatchSnapshot()
})
