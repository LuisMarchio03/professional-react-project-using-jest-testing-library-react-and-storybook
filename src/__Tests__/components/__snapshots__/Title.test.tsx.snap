// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<DocumentFragment>
  <h1
    class="c-eUWuOM"
  >
    Title
  </h1>
</DocumentFragment>
`;
